pedersen/src/pedersen.rs

```diff
@@ -0,0 +1,118 @@
+use crate::hash_context::get_standard_pedersen_hash_context;
+use ark_ff::{BigInt, BigInteger, Field, PrimeField};
+use felt::Felt252;
+use num_bigint::BigUint;
+
+pub struct PedersenHash {
+    state: Felt252,
+}
+
+impl PedersenHash {
+    pub fn init_digest_to(digest: Vec<u8>) -> Self {
+        PedersenHash {
+            state: bytes_to_field(&digest),
+        }
+    }
+
+    pub fn get_digest(&self) -> Vec<u8> {
+        self.state.into_bigint().to_bytes_be()
+    }
+
+    pub fn hash(val0: PedersenHash, val1: PedersenHash) -> PedersenHash {
+        let cntx = get_standard_pedersen_hash_context();
+        let res = cntx.hash_elements(val0.state, val1.state);
+        PedersenHash { state: res }
+    }
+
+    pub fn hash_bytes_with_length(bytes: Vec<u8>) -> PedersenHash {
+        let mut state = Felt252::ZERO;
+        let cntx = get_standard_pedersen_hash_context();
+        let modulus = BigUint::from_bytes_be(&Felt252::MODULUS.to_bytes_be());
```

Comment:
I am using `BigUint` (https://docs.rs/num-bigint/latest/num_bigint/struct.BigUint.html) and not `Bigint<N>` because `BigUint` has functions for `/` and `%`.